src/core/js/views/buttonsView.js

```diff
@@ -76,48 +72,51 @@ define(function() {
-                    case "showCorrectAnswer": case "hideCorrectAnswer":
-    				    //make model answer button inaccessible but enabled for visual users
-    				    //	due to inability to represent selected incorrect/correct answers to a screen reader, may need revisiting
-                        this.$('.buttons-action').a11y_cntrl(false).html(buttonText).attr('aria-label', ariaLabel);
-                        break;
-                    default:
-    				    //enabled button, make accessible and update aria labels and text.
-                        this.$('.buttons-action').a11y_cntrl_enabled(true).html(buttonText).attr('aria-label', ariaLabel);
-                }
+          } else {
+            // Backwords compatibility with v1.x
+            var ariaLabel = this.model.get('_buttons')["_" + changedAttribute].ariaLabel;
+            var buttonText = this.model.get('_buttons')["_" + changedAttribute].buttonText;
+
+            switch (changedAttribute) {
+              case "submit":
+              case "showCorrectAnswer":
+              case "hideCorrectAnswer":
+                //make model answer button inaccessible but enabled for visual users
+                //	due to inability to represent selected incorrect/correct answers to a screen reader, may need revisiting
+                this.$('.buttons-action').a11y_cntrl_enabled(true).html(buttonText).attr('aria-label', ariaLabel);
+                break;
+              default:
+                //enabled button, make accessible and update aria labels and text.
+                this.$('.buttons-action').a11y_cntrl_enabled(false).html(buttonText).attr('aria-label', ariaLabel);
+                break;
             }
+          }
 
-            this.updateAttemptsCount();
+          this.updateAttemptsCount();
         },
 
         checkFeedbackState: function(){
-            if (!this.model.get('_canShowFeedback')) {
-                // If feedback should be hidden, the 'Submit' button should stretch
-                // to fill the space and the 'Feedback' button should be hidden.
-                // (These classes must be implemented in the theme.)
-                this.$('.buttons-action').addClass('buttons-action-fullwidth');
-                this.$('.buttons-feedback').addClass('no-feedback');
-            }
+          // If feedback should be hidden, the 'Submit' button should stretch
+          // to fill the space and the 'Feedback' button should be hidden.
+          // (These classes must be implemented in the theme.)
+          this.$('.buttons-action').toggleClass('buttons-action-fullwidth', !this.model.get('_canShowFeedback'));
+          this.$('.buttons-action').toggleClass('buttons-action-enlarge', !this.model.get('_canShowFeedback'));
+          this.$('.buttons-feedback').toggleClass('no-feedback', !this.model.get('_canShowFeedback'));
+          this.$('.buttons-marking-icon').toggleClass('no-feedback', !this.model.get('_canShowFeedback'));
```

Comment:
Could we reuse variables here rather than doing multiple lookups?

For example:
```js
var canShowFeedback = this.model.get('_canShowFeedback');

this.$('.buttons-action').toggleClass('buttons-action-fullwidth buttons-action-enlarge', !canShowFeedback);
this.$('.buttons-feedback').toggleClass('no-feedback', !canShowFeedback);
this.$('.buttons-marking-icon').toggleClass('no-feedback', !canShowFeedback);
```